#ifndef VULNERABILITYEDITORWIDGET_H
#define VULNERABILITYEDITORWIDGET_H

#include <QWidget>
#include "gamedata/LaylaVulnerabilitySettings.h"

namespace Ui {
class VulnerabilityEditorWidget;
}

class VulnerabilityEditorWidget : public QWidget
{
    Q_OBJECT
    
public:
    explicit VulnerabilityEditorWidget(QWidget *parent = 0);
    ~VulnerabilityEditorWidget();
    
    void editSettings(Lonely::LaylaVulnerabilitySettings& settings__);

    bool settingsLoaded() const;

protected:
    Lonely::LaylaVulnerabilitySettings* settings_;

private slots:
    void on_pistolBox_clicked(bool checked);

    void on_machineGunBox_clicked(bool checked);

    void on_grenadeBox_clicked(bool checked);

    void on_flamethrowerBox_clicked(bool checked);

    void on_axeBox_clicked(bool checked);

    void on_bazookaBox_clicked(bool checked);

    void on_saberBox_clicked(bool checked);

    void on_shieldBox_clicked(bool checked);

private:
    Ui::VulnerabilityEditorWidget *ui;
};

#endif // VULNERABILITYEDITORWIDGET_H
